packages/guilded.js/lib/gateway/handler/TeamEventHandler.ts

```diff
@@ -1,8 +1,20 @@
 import type { WSTeamRolesUpdatedPayload } from "@guildedjs/guilded-api-typings";
+import { constants } from "../../constants";
 import { GatewayEventHandler } from "./GatewayEventHandler";
 
 export class TeamEventHandler extends GatewayEventHandler {
     teamRolesUpdated(data: WSTeamRolesUpdatedPayload): boolean {
-        return false;
+        const newMembers = [];
+        const oldMembers = [];
+        for (const m of data.d.memberRoleIds) {
+            const member = this.client.members.cache.get(`${data.d.serverId}:${m.userId}`);
+            if (!member) {
+                newMembers.push(m);
+                continue;
+            }
+            oldMembers.push(member._clone());
+            newMembers.push(member._update({ roleIds: m.roleIds }));
+        }
+        return this.client.emit(constants.clientEvents.TEAM_ROLES_UPATED, data.d.serverId, newMembers, oldMembers)
```

Comment:
TeamMembers have a serverId property on them already, so emitting one is extraneous

Reply:
did this because there wouldn't be a serverId property if the member isn't cached, but I see you've solved that with the next suggestion, will fix in a bit!